components/pd_client/src/util.rs

```diff
@@ -197,7 +201,9 @@ impl LeaderClient {
             }
             inner.hb_sender = Either::Left(Some(tx));
             if let Either::Right(ref mut task) = inner.hb_receiver {
```

Comment:
How about `mem::replace(&mut inner.hb_receiver, Either::Left(Some(rx))`? So that the option can be removed.